File: employeewebapp/src/main/java/com/exl/employeewebapp/entity/Employee.java
package com.exl.employeewebapp.entity;

import java.time.LocalDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name="employee")
public class Employee {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column(name = "employee_id")
	private int id;
	@Column(name = "first_name")
	private String firstName;
	@Column(name = "last_name")
	private String lastName;
	@Column(name = "job_title")
	private String jobTitle;
	@Column(name = "age")
	private int age;
	@Column(name = "start_date")
	private LocalDate startDate;
	@Column(name = "end_date")
	private LocalDate endDate;
	
	public Employee() {
		
	}
	
	public Employee(String firstName, String lastName, String jobTitle, int age, LocalDate startDate, LocalDate endDate) {
		super();
		this.firstName = firstName;
		this.lastName = lastName;
		this.jobTitle = jobTitle;
		this.age = age;
		this.startDate = startDate;
		this.endDate = endDate;
	}

	public String fullName() {
		return String.format("%s %s",
				firstName, lastName);
	}
	
	
	public int getId() {
		return id;
	}
	
	public void setId(int id) {
		this.id = id;
	}
	
	public String getFirstName() {
		return firstName;
	}
	
	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}
	
	public String getLastName() {
		return lastName;
	}
	
	public void setLastName(String lastName) {
		this.lastName = lastName;
	}
	
	public String getJobTitle() {
		return jobTitle;
	}

	public void setJobTitle(String jobTitle) {
		this.jobTitle = jobTitle;
	}

	public int getAge() {
		return age;
	}
	
	public void setAge(int age) {
		this.age = age;
	}
	
	public LocalDate getStartDate() {
		return startDate;
	}
	
	public void setStartDate(LocalDate startDate) {
		this.startDate = startDate;
	}
	
	public LocalDate getEndDate() {
		return endDate;
	}
	
	public void setEndDate(LocalDate endDate) {
		this.endDate = endDate;
	}
	
	
}

File: angular-employee/src/app/common/employee.ts
export class Employee {
    id: number;
    firstName: string;
    lastName: string;
    jobTitle: string;
    age: number;
    startDate: Date;
    endDate: Date;

    fullName(): string{
        return this.firstName + " " + this.lastName;
    }
    
    public isInFullName(name: string): boolean{
        //console.log("Validating Name");
        if(name == null) 
            return true;
        
        return (this.fullName().includes(name));
      }
    
    public isWithinDate(start: Date, end: Date): boolean{
        if( start == null && end == null )
            return true;
        
        if(start == null)
            return this.endDate < end;

        if(end == null)
            return this.startDate > start;
            
        return this.startDate > start && this.endDate < end;
    }
}


